go.mod

```diff
@@ -1,6 +1,6 @@
 module kpt.dev/configsync
 
-go 1.22.0
+go 1.23
```

Comment:
Is this urgent? I have an action item to write a playbook for updating the Go version which I haven't gotten around to yet. The Go version needs to be updated in the build environment in a couple places before it's updated here.

Reply:
It's blocking https://github.com/GoogleContainerTools/kpt-config-sync/pull/1516, which uses `maps.Keys`. but if we need to hold off i could revert that change.